### PR TITLE
architecture(sw360WorkflowSteps): separating Implementation of SW360 workflowsteps from AntennaContext

### PR DESCRIPTION
**Pull request by neubs-bsi**

### CODE REVIEW

**Comment by blaumeiser-at-bosch** [comment]
Azure build red: https://dev.azure.com/EclipseSW360antenna/EclipseSW360antenna/_build/results?buildId=359

**Comment by maxhbr** [comment]
the build 359 points to a build for the policy engine pr

**Comment by maxhbr** [comment]
Build is red due to PMD violation

**Comment by maxhbr** [comment]
still failing due to PMD violations:
```
[INFO] PMD Failure: org.eclipse.sw360.antenna.sw360.workflow.processors.SW360Enricher:20 Rule:UnusedImports Priority:4 Avoid unused imports such as 'org.slf4j.Logger'.
[INFO] PMD Failure: org.eclipse.sw360.antenna.sw360.workflow.processors.SW360Enricher:21 Rule:UnusedImports Priority:4 Avoid unused imports such as 'org.slf4j.LoggerFactory'.
```

see: https://dev.azure.com/EclipseSW360antenna/EclipseSW360antenna/_build/results?buildId=363

**Comment by neubs-bsi** [comment]
waiting for #262 to be merged first

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/processors/SW360EnricherImpl.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+        LOGGER.info("Artifact: '" + String.join("', '", artifactLicenseNames)
+                + "' <-> SW360: '" + String.join("', '", releaseLicenseNames) + "'");
+    }
+
+    private void addCPEIdIfAvailable(Artifact artifact, SW360Release release) {
+        final String CPE_PREFIX = "cpe:2.3:";
+        final String OLD_CPE_PREFIX = "cpe:/";
+        String cpeId = release.getCpeid();
+        if (cpeId != null && (cpeId.startsWith(CPE_PREFIX) || cpeId.startsWith(OLD_CPE_PREFIX))) {
```
Comment (maxhbr) [inline modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/processors/SW360EnricherImpl.java]: maybe better:
```suggestion
        if (cpeId != null && (cpeId.startsWith("cpe:"))) {
```

File: modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        this.sw360ProxyUse = sw360ProxyUse;
+        this.sw360ProxyHost = sw360ProxyHost;
+        this.sw360ProxyPort = sw360ProxyPort;
+        this.projectName = projectName;
+        this.projectVersion = projectVersion;
+        this.updateReleases = updateReleases;
+    }
+
+    public Map<String, IAttachable> produce(Collection<Artifact> intermediates) throws AntennaException {
+        SW360MetaDataUpdater sw360MetaDataUpdater = new SW360MetaDataUpdater(
```
Comment (maxhbr) [inline modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360UpdaterImpl.java]: If you intantiate the `SW360MetaDataUpdater` in the constructor, you might be able to get rid of most of the fields

**Pull request merged**